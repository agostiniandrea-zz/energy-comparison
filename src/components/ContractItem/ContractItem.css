.contract-item {
    position: relative;
    border: 2px solid grey;
    border-radius: 15px;
    margin-bottom: 5px;
    height: 100px;
}

.contract-container {
    position: absolute;
    padding: 15px;
}

.brand-logo {
    float: left;
    width: 80px;
    height: 44px;
    margin-top: 15px;
}

.brand-name {
    display: inline-flex;
    float: left;
    margin-top: 10px;
    margin-left: 10px;
}

.contract-price {
    position: relative;
    display: inline-flex;
    text-align: left;
    margin-top: 15px;
    font-size: 20px;
    font-weight: bold;
}

.description {
    margin-left: 10px;
    font-size: 9px;
    font-weight: normal;
    text-align: left;
}

/* Smartphones (portrait and landscape) ----------- */
@media only screen and (min-device-width : 320px) {
    .contract-price {
        position: absolute;
        left: 105px;
        width: 200px;
        top: 35px;
    }
}

/* iPads (portrait and landscape) ----------- */
@media only screen and (min-device-width : 768px) {
    .brand-name {
        margin-top: 30px;
        margin-left: 20px;
    }

    .contract-price {
        position: absolute;
        left: 400px;
        margin-top: 5px;
        min-width: 180px;
    }
}

@media only screen and (min-width : 1024px) {
    .brand-name {
        margin-left: 100px;
    }

    .contract-price {
        left: 650px;
        min-width: 240px;
    }
}

@media only screen and (min-width : 1366px) {
    .contract-price {
        left: 950px;
    }
}