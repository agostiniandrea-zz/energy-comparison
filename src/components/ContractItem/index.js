import React from 'react';
import './ContractItem.css';

class ContractItem extends React.Component {
    constructor(props) {
        super(props);

        this.annualPrice = this.annualPrice.bind(this);
        this.monthlyPrice = this.monthlyPrice.bind(this);
    }

    annualPrice() {
        return ((this.props.price * this.props.quantity) / 10000).toFixed(2);
    }

    monthlyPrice() {
        return (this.annualPrice() / 12).toFixed(2);
    }

    render() {
        return (
            <div className="contract-item">
                <div className="contract-container">
                    <img className="brand-logo" alt="" src={this.props.image} />
                    <div className="brand-name">{this.props.name}</div>
                    <div className="contract-price">
                        <div className="price">€ {this.monthlyPrice()}</div>
                        <label className="description">
                            <strong>per month</strong><br />
                            € {this.annualPrice()} per year
                        </label>
                    </div>
                </div>
            </div>
        );
    }
}

export default ContractItem;
