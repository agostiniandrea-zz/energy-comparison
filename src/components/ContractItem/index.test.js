import React from 'react';
import { shallow } from 'enzyme';
import ContractItem from './index';

import gasProvidersList from './../../mock/gas.json';
const providersList = gasProvidersList.list;
const provider = providersList[0];

const QUANTITY_VALUE = 500;

let wrapper1;
let wrapper2;

beforeEach(() => {
    wrapper1 = shallow(<ContractItem />);
    wrapper2 = shallow(<ContractItem image={provider.image} name={provider.name} price={provider.price} quantity={QUANTITY_VALUE} />);
})

afterEach(() => {
    wrapper1.unmount();
    wrapper2.unmount();
});

describe('ContractItem/index.js', () => {
    it('renders without crashing', () => {
        expect(wrapper1).toMatchSnapshot();
        expect(wrapper2).toMatchSnapshot();
    });

    it('calculates prices properly', () => {
        expect(wrapper2.instance().annualPrice()).toBe("14.10");
        expect(wrapper2.instance().monthlyPrice()).toBe("1.18");
    });
});