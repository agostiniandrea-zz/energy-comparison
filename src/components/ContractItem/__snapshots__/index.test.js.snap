// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContractItem/index.js renders without crashing 1`] = `ShallowWrapper {}`;

exports[`ContractItem/index.js renders without crashing 2`] = `ShallowWrapper {}`;
